gcs/constraints/smart_table.cc

```diff
@@ -90,31 +82,62 @@ namespace
 
                 switch (binary_entry.constraint_type) {
                 case ConstraintType::LESS_THAN:
+                    if (state.maybe_proof()) {
```

Comment:
OK, so for this and all the other calls to log_filtering_inference: I think you are outputting proof steps even if no filtering ends up being done? I tried the following dumb instance:

```
    auto x = p.create_integer_variable(-1_i, 3_i, "x");
    auto z = p.create_integer_variable(-1_i, 3_i, "z");
    auto y = p.create_integer_variable(-1_i, 3_i, "y");
    auto tuples = SmartTuples{ {NotEqualsVar{x, y}} };
    p.post(SmartTable{{x, y}, tuples});
    p.post(Equals{x, 3_c});
```

which I would expect to log one inference, once, and then not do anything again. But it looks like it's logging `u guesses 1 ~flag0_t0 1 ~iv2_eq_3 >= 1 ;` every time the propagator is called, even if no inference is done?

Reply:
I guess this might be why you have massive proofs?

Reply:
That's a good point. I think I've fixed it now

Reply:
Proof sizes are still quite large but that makes a significant improvement. For largest smart table on 6 variables reduced to 96,020 lines from 136,314